File: user-profile/src/main/java/com/oxygensened/userprofile/infrastructure/storage/StorageConfiguration.java
package com.oxygensened.userprofile.infrastructure.storage;

import com.oxygensend.commonspring.storage.StorageService;
import com.oxygensened.userprofile.application.storage.ThumbnailService;
import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@EnableConfigurationProperties(StorageProperties.class)
@Configuration
public class StorageConfiguration {

    @Bean
    ThumbnailService thumbnailCreator(StorageService storageService, StorageProperties storageProperties) {
        return new ScalrThumbnailService(storageService, storageProperties.thumbnailDir());
    }
}

File: static-data/src/main/kotlin/com/oxygensend/staticdata/infrastructure/storage/StorageConfiguration.kt
package com.oxygensend.staticdata.infrastructure.storage

import com.oxygensend.commonspring.storage.StorageService
import com.oxygensend.staticdata.application.storage.ImageService
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@EnableConfigurationProperties(StorageProperties::class)
@Configuration
class StorageConfiguration {
    @Bean
    fun aboutUsImageService(storageService: StorageService?, storageProperties: StorageProperties): ImageService {
        return ImageServiceImpl(storageService!!, storageProperties.aboutUsImageDir!!)
    }
}
